code/modules/mob/living/basic/space_fauna/revenant/_revenant.dm

```diff
@@ -0,0 +1,499 @@
+
+/mob/living/basic/revenant/Destroy()
+	GLOB.revenant_relay_mobs -= src
+	return ..()
+
+/mob/living/basic/revenant/Login()
+	. = ..()
+	if(!. || isnull(client))
+		return FALSE
+
+	var/static/cached_string = null
+	if(isnull(cached_string))
+		cached_string = examine_block(jointext(create_login_string(), "\n"))
+
+	to_chat(src, cached_string, type = MESSAGE_TYPE_INFO)
+
+	if(generated_objectives_and_spells)
+		return TRUE
+
+	generated_objectives_and_spells = TRUE
+	mind.set_assigned_role(SSjob.GetJobType(/datum/job/revenant))
+	mind.special_role = ROLE_REVENANT
+	SEND_SOUND(src, sound('sound/effects/ghost.ogg'))
+	mind.add_antag_datum(/datum/antagonist/revenant)
+	return TRUE
+
+/// Signal Handler Injection to handle Life() stuff for revenants
+/mob/living/basic/revenant/proc/on_life(seconds_per_tick = SSMOBS_DT, times_fired)
+	SIGNAL_HANDLER
+
+	if(dormant)
+		return COMPONENT_LIVING_CANCEL_LIFE_PROCESSING
+
+	if(revealed && essence <= 0)
+		INVOKE_ASYNC(src, PROC_REF(death))
+		return COMPONENT_LIVING_CANCEL_LIFE_PROCESSING
+
+	if(unreveal_time && world.time >= unreveal_time)
+		unreveal_time = 0
+		revealed = FALSE
+		incorporeal_move = INCORPOREAL_MOVE_JAUNT
+		invisibility = INVISIBILITY_REVENANT
+		to_chat(src, span_revenboldnotice("You are once again concealed."))
+
+	if(unstun_time && world.time >= unstun_time)
+		unstun_time = 0
+		REMOVE_TRAIT(src, TRAIT_NO_TRANSFORM, REVENANT_STUNNED_TRAIT)
+		to_chat(src, span_revenboldnotice("You can move again!"))
+
+	if(essence_regenerating && !inhibited && essence < max_essence) //While inhibited, essence will not regenerate
+		var/change_in_time = DELTA_WORLD_TIME(SSmobs)
+		essence = min(essence + (essence_regen_amount * change_in_time), max_essence)
+		update_mob_action_buttons() //because we update something required by our spells in life, we need to update our buttons
+
+	update_appearance(UPDATE_ICON)
+	update_health_hud()
+
+/mob/living/basic/revenant/get_status_tab_items()
+	. = ..()
+	. += "Current Essence: [essence >= max_essence ? essence : "[essence] / [max_essence]"] E"
+	. += "Total Essence Stolen: [essence_accumulated] SE"
+	. += "Unused Stolen Essence: [essence_excess] SE"
+	. += "Perfect Souls Stolen: [perfectsouls]"
+
+/mob/living/basic/revenant/update_health_hud()
+	if(isnull(hud_used))
+		return
+
+	var/essencecolor = "#8F48C6"
+	if(essence > max_essence)
+		essencecolor = "#9A5ACB" //oh boy you've got a lot of essence
+	else if(essence <= 0)
+		essencecolor = "#1D2953" //oh jeez you're dying
+	hud_used.healths.maptext = MAPTEXT("<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='[essencecolor]'>[essence]E</font></div>")
+
+/mob/living/basic/revenant/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterproof = null, message_range = 7, datum/saymode/saymode = null)
+	if (!message)
+		return
+
+	if (src.client)
+		if(client.prefs.muted & MUTE_IC)
```

Comment:
For consistency, either drop the `src.` in client or use `src.` in the `client.prefs.muted` line